File: mcorg.frontends/organizer-webapp/api/resources/models/ResourcePackResponse.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { ResourceResponse } from './ResourceResponse';

export type ResourcePackResponse = {
  id: string;
  name: string;
  version: string;
  serverType: ResourcePackResponse.serverType;
  resources: Array<ResourceResponse>;
};

export namespace ResourcePackResponse {

  export enum serverType {
    FABRIC = 'FABRIC',
    FORGE = 'FORGE',
  }


}

File: mcorg.backends/resource-service/src/main/java/app/mcorg/resources/presentation/entities/ResourcePackResponse.java
package app.mcorg.resources.presentation.entities;

import app.mcorg.resources.domain.model.resource.ResourcePack;
import app.mcorg.resources.domain.model.resource.ServerType;
import org.springframework.lang.NonNull;

import java.util.List;

public record ResourcePackResponse(@NonNull String id,
                                   @NonNull String name,
                                   @NonNull String version,
                                   @NonNull ServerType serverType,
                                   @NonNull List<ResourceResponse> resources) {
    public static ResourcePackResponse from(ResourcePack pack) {
        return new ResourcePackResponse(
                pack.getId(),
                pack.getName(),
                pack.getVersion(),
                pack.getServerType(),
                pack.getResources().stream()
                        .map(ResourceResponse::from)
                        .toList()
        );
    }
}
